qemu/tests/numa_consistency.py

```diff
@@ -1,125 +1,99 @@
-        node_used_host = get_vcpu_used_node(host_numa_node,
-                                            vcpu_threads[cpuid])
-        node_used_host_index = node_list.index(node_used_host)
-        memory_used_before = memory_status[node_used_host_index]
-        error_context.context("Allocate memory in guest", test.log.info)
-        session.cmd(mount_cmd)
-        binded_dd_cmd = "taskset %s" % str(2 ** int(cpuid))
-        binded_dd_cmd += " dd if=/dev/urandom of=/tmp/%s" % cpuid
-        binded_dd_cmd += " bs=1M count=%s" % dd_size
-        session.cmd(binded_dd_cmd)
-        error_context.context("Check qemu process memory use status",
-                              test.log.info)
-        node_after = get_vcpu_used_node(host_numa_node, vcpu_threads[cpuid])
-        if node_after != node_used_host:
-            test.log.warn("Node used by vcpu thread changed. So drop the"
-                          " results in this round.")
-            drop += 1
-            continue
-        memory_status, _ = utils_test.qemu.get_numa_status(host_numa_node,
-                                                           qemu_pid)
-        memory_used_after = memory_status[node_used_host_index]
-        page_size = resource.getpagesize() / 1024
-        memory_allocated = (memory_used_after -
-                            memory_used_before) * page_size / 1024
-        if 1 - float(memory_allocated) / float(dd_size) > threshold:
-            numa_hardware_cmd = params.get("numa_hardware_cmd")
-            if numa_hardware_cmd:
-                numa_info = process.system_output(numa_hardware_cmd,
-                                                  ignore_status=True,
-                                                  shell=True)
-            msg = "Expect malloc %sM memory in node %s," % (dd_size,
-                                                            node_used_host)
-            msg += "but only malloc %sM \n" % memory_allocated
-            msg += "Please check more details of the numa node: %s" % numa_info
-            test.fail(msg)
-    session.close()
-
-    if drop == len(vcpu_threads):
-        test.error("All test rounds are dropped. Please test it again.")
+    try:
+        test_mem = float(params.get("mem")) * mem_ratio
+        guest_stress_args = params.get("guest_stress_args", "-a -p -l %sM")
+        guest_stress_args = guest_stress_args % int(test_mem)
+        stress_path = os.path.join(data_dir.get_deps_dir('mem_mapping'),
+                                   mem_map_tool)
+        utils_memory.drop_caches()
+        for node in node_list:
+            error_context.base_context("Get the qemu memory use for node: %d before stress"
+                                       % node_alloc, test.log.info)
+            memory_before = get_node_used_memory(qemu_pid, node_alloc)
+            try:
+                error_context.context("Executing stress test in node: %d"
+                                      % node, test.log.info)
```

Comment:
What's the relationship for the stress in vm and the node in host ?

Reply:
@yanan-fu basically the QEMU process is binded to one host node, so executing the stress inside the VM will create a high memory usage in that node. The error_context trace is merely informative, to let everybody know the node it's being used.

Reply:
But, here is a loop for available nodes in host while the qemu process be bind to the specified host node (`node_alloc`).  I can not see any relationship for the stress and node with vm here.

Reply:
@yanan-fu got your point, it's not necessary but with the loop could be observed more than once how memory increase in the binded node. Do you consider it'd easier/better removing this section?

Reply:
The simpler, the better, removed the loop